pkg/logs/input/container/scanner.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package container
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/input/docker"
+	"github.com/DataDog/datadog-agent/pkg/logs/input/kubernetes"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NewScanner returns a new container scanner,
+// by default it returns a docker scanner unless it could not be set up properly ,
```

Comment:
- extra space before the trailing ` ,`
- after `unless` I'd expect a `in which case` instead of `otherwise` :)
- It seems very important to explain why we try to use docker before kubernetes. Let's please add a detailed comment on this.